datafusion/expr/src/udf.rs

```diff
@@ -342,6 +348,56 @@ pub struct ScalarFunctionArgs<'a> {
     pub return_type: &'a DataType,
 }
 
+#[derive(Debug)]
+pub struct ReturnTypeArgs<'a> {
+    /// The data types of the arguments to the function
+    pub arg_types: &'a [DataType],
+    /// The Utf8 arguments to the function, if the expression is not Utf8, it will be empty string
+    pub arguments: &'a [Option<&'a ScalarValue>],
```

Comment:
I also recommend naming this `scalar_arguments`

```suggestion
    pub scalar_arguments: &'a [Option<&'a ScalarValue>],
```